app/helpers/katello/katello_urls_helper.rb

```diff
@@ -6,12 +6,8 @@ def foreman_settings_url(schema = 'http')
       ::Setting[:foreman_url].sub(%r|^.*?:|, "#{schema}:")
     end
 
-    def subscription_manager_configuration_url(host = nil, rpm = true)
-      prefix = if host&.content_source
-                 "http://#{host.content_source.hostname}"
-               else
-                 foreman_settings_url
-               end
+    def subscription_manager_configuration_url(host = nil, rpm = true, prefix = nil)
+      prefix ||= host&.content_source ? "http://#{host.content_source.hostname}" : foreman_settings_url
```

Comment:
Turns out we'll need to do this slightly differently. We need to only get the hostname part of the url and use that instead of host&.content_source.hostname. Something like this

```ruby
def subscription_manager_configuration_url(host = nil, rpm = true, hostname = nil)
  prefix = if host&.content_source
￼                 "http://#{hostname || host.content_source.hostname}"
￼               else
￼                 foreman_settings_url
￼               end

...
```
My appologies for sending you the wrong direction first. We can do the parsing either in the template or here, but we must only take the hostname to ensure, the resulting url using http.

Reply:
one clarification: the logic in fact is, if hostname is set, use that with http://, otherwise remains

Reply:
Maybe we shouldn't expect just `hostname`, but parse it from the `url argument`,
like this:
```ruby
def subscription_manager_configuration_url(host = nil, rpm = true, url = nil)
if url
  prefix = "http://#{URI(url).host}"
else
  prefix = host&.content_source ? "http://#{host.content_source.hostname}" : foreman_settings_url
end
```
That way users won't have to parse the URL manually. Thoughts?

Reply:
After a short discussion, it seems better to take just hosname here, the extraction should happen in the controller that renders the template. There are more urls and hostnames we'll need to prepare to variables prior the template rendering.